### PR TITLE
Remove rounded corners and accessibility highlights on reporting column headers

### PR DESCRIPTION
**Pull request by gilluminate**
Closes #<issue>

### Description Of Changes

Takes a few scattered styles for defining headers/cells and combines them a little better. Also removes the sorting for now.

### Code Changes

* [x] update the `getTableTHandTDStyles` util to include headers that act as buttons
* [x] add smarter padding definitions to `getTableTHandTDStyles` using Chakra's spacing definitions (for consistency) as well as a `calc` to account for the width of the column resizer
* [x] take advantage of Chakra's `sx` function as opposed to hard coding the `style` property in several instances.
* [x] update table header menu button size to match the table header, preventing it from causing height inconsistencies.
* [x] add focus state to table header menu button to preserve accessibility for keyboard users now that the outline is no longer visible.

### Steps to Confirm

* [ ] `/consent/properties` has a table with 3 types of headers. It's a good one to look at for styling consistency.

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded(https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`(https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/admin-ui/src/features/common/table/v2/FidesTable.tsx [hunk 1]
```diff
@@ -26,7 +26,8 @@ import React, { ReactNode, useMemo, useState } from "react";
 
 import { DisplayAllIcon, GroupedIcon } from "~/features/common/Icon";
 import { FidesRow } from "~/features/common/table/v2/FidesRow";
-import { getTableTHandTDStyles } from "~/features/common/table/v2/util";
+
+import { getTableTHandTDStyles } from "./util";
```
Comment (Kelsey-Ethyca) [inline clients/admin-ui/src/features/common/table/v2/FidesTable.tsx:30]:
```suggestion
import { getTableTHandTDStyles } from "~/features/common/table/v2/util";
```

File: clients/admin-ui/src/features/common/table/v2/FidesTable.tsx [hunk 2]
```diff
@@ -73,10 +77,15 @@ const HeaderContent = <T,>({
...
+        size="sm"
+        height={9} // same as table header height
         width="100%"
-        pr={1}
+        sx={{ ...getTableTHandTDStyles(header.column.id) }}
         textAlign="start"
         data-testid={`${header.id}-header-menu`}
+        _focus={{
+          backgroundColor: "gray.100",
+        }}
```
Comment (Kelsey-Ethyca) [inline clients/admin-ui/src/features/common/table/v2/FidesTable.tsx:88]:
```suggestion
        _focus={{
          outline: "none",
        }}
```

**Review by Kelsey-Ethyca — CHANGES_REQUESTED** [review]
Let me know what you think of these suggestions. UAT passing!

File: clients/admin-ui/src/features/common/table/v2/FidesTable.tsx (hunk 2 above)
Comment (gilluminate) [reply clients/admin-ui/src/features/common/table/v2/FidesTable.tsx:88]: with this change, a keyboard user has no indication where the focus is, which is why I added bg color, which matches the hover state. Could still hide the outline.

```suggestion
        _focus={{
          backgroundColor: "gray.100",
          outline: "none",
        }}
```

File: clients/admin-ui/src/features/common/table/v2/FidesTable.tsx (hunk 1 above)
Comment (gilluminate) [reply clients/admin-ui/src/features/common/table/v2/FidesTable.tsx:30]: Is this a preferred pattern, to have the more full path? VS Code leans toward shortest, so I will keep an eye on that.

Comment (Kelsey-Ethyca) [reply clients/admin-ui/src/features/common/table/v2/FidesTable.tsx]: yes we prefer the absolute path

**Review by Kelsey-Ethyca — APPROVED** [review]

**Pull request merged**